lesson/models.py

```diff
@@ -0,0 +1,18 @@
+from django.db import models
+from tinymce.models import HTMLField
+
+class Subset(models.Model):
+    name = models.CharField(max_length=120)
+
+    def __str__(self):
+        return self.name
+
+class Lesson(models.Model):
+    title = models.CharField(max_length=120)
+    subset = models.ForeignKey(Subset, on_delete=models.CASCADE)
```

Comment:
Something interesting with this subset seems to be happening.  On the admin page, everything looks fine: both the subset and lesson APIs.  However, when I call the content using the GET method, I get a response that I'm not expecting for the subset.  I think looks like the subset is being saved as the ID.  Instead of models.ForeignKey, shouldn't this be models.CharField?  And shouldn't there be a separate field for lesson ID where you use ForeignKey?

![image](https://github.com/YomenT/bluebird-teaching/assets/32947641/1ab78460-38e9-425b-b353-36640ae2c3a2)

Not saying this is like wrong.  Maybe there is something you're doing that I'm not seeing on the surface, but I would have expected the "subset" field to be something like "Heart of Algebra", and then there would be a separate field that's just like "id" or something like that.

Reply:
There are a few reasons why I used a ForeignKey instead of a CharField. Firstly, I wanted to create a dropdown feature where the subset model would be linked to the lesson's model to avoid any differences in the text formatting (uppercase, lowercase, etc.) If you look at the subset model, you will see the CharField present there. Another reason was to make it simpler to query where you select the lessons based on if the ID matches. However, there is a way to retain this functionality while showing the actual name of the subset in the JSON response without requiring an additional field. I will implement this and update the PR.

Reply:
Gotcha I see what you're saying.  Thanks for pointing that out.  All that is totally fine then.  If you haven't done the update yet, I would actually want to hold off so we can discuss this.  The solution that you mentioned is totally cool.  I just want to talk this over.  There are a couple of solutions, and I just want to weigh the pros and cons with you inshallah.  I feel like it's a good conversation to have.